product_combination_exclude/models/product_product.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Graeme Gellatly
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = 'product.product'
+
+    @api.multi
+    def create(self, vals):
+        vals = dict(vals or {})
```

Comment:
I would love to explain to you why they do this, but instead I'm going to shut down like you did. You're wrong.